http/server.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2022 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package http
+
+import (
+	"fmt"
+	"github.com/spacemonkeygo/monkit/v3/present"
+	"net/http"
+
+	"github.com/spacemonkeygo/monkit/v3"
+)
+
+// TraceHandler wraps a HTTPHandler and import trace information from header.
+func TraceHandler(c http.Handler, scope *monkit.Scope) http.Handler {
+	return traceHandler{
+		handler: c,
+		scope:   scope,
+	}
+}
+
+type traceHandler struct {
+	handler http.Handler
+	scope   *monkit.Scope
+}
+
+// ServeHTTP implements http.Handler with span propagation.
+func (t traceHandler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
+
+	info := TraceInfoFromHeader(request.Header)
+
+	traceId := monkit.NewId()
+	if info.TraceId != nil {
+		traceId = *info.TraceId
+	}
+
+	trace := monkit.NewTrace(traceId)
+	ctx := request.Context()
+
+	parent := int64(0)
+	if info.ParentId != nil {
+		parent = *info.ParentId
+	}
+
+	if info.Sampled {
+		trace.Set(present.SampledKey, true)
```

Comment:
can you also check for SampledCBKey and call that callback if it's set?

Reply:
Can you please help me to understand the motivation behind `SampledCBKey`. Is there any good example for the usage? When is it supposed to be called? Before sending out the trace? After the creation?

Just not sure what should I do here. The trace is created a few lines before, and nobody can set the `SampledCBKey` until this line.

The first point when it might be set is `t.scope.Func().RemoteTrace(&ctx, parent, trace)` where the other trace observers are called. But when should I call the `cb`. After or before `t.handler.ServeHTTP(wrapped, request.WithContext(s))` ?

Reply:
SampledCBKey is used for setting a callback that fires whenever a new trace goes from unsampled to sampled. the point with SampledCBKey is that you can register your own local collector that collects sampled traces this way. if you look in present for where SampledKey is also set, that is all that you need to do here too

Reply:
Ok, I added 

```
if cb, exists := trace.Get(present.SampledCBKey).(func(*monkit.Trace)); exists {
		cb(trace)
	}
```

But still don't understand why is it good for. Where is it used?

In this special case I create the `Trace` so I am 100% sure, that there is no callback registered (`Trace` is not context).

The only way to add a callback is register a `TraceObserver` as it's called by `t.scope.Func().RemoteTrace(&ctx, parent, trace)`

But if you already have a `TraceObserver` which is called, why is it good to register a `cb` instead of just doing whatever you would like to do?

Reply:
oh you're right, okay i'm dumb.